### PR TITLE
administrator manage resources

### PR DESCRIPTION
**Pull request by privacywill**
Project administrator create resources for each developer, as a result the developers only need to build and push the image and deploy. As a result, the developers only require `Artifact registry writer` role in gcp project.

Related issue: #9 

# Changes

* Rename the `reousrce/gcp` to `resource/global`.
* Rename the `reousrce/kubernetes` to `resource/developer-specific`.
* Add a script `push_image.sh` to push the image. Because the repository parameters is too long in bazel oci_push cmdline. 
* Create developer specific database and db account.  Move it to `resource/developer-specific`.

# Testing

Manual end-to-end testing.

### CODE REVIEW

**Comment by dayeol** [comment]
can you try to use `multirun` instead of a bash script?
See #7

**Review by dayeol — CHANGES_REQUESTED** [review]

File: deployment/jupyterhub/deploy.sh [hunk 1]
```diff
@@ -30,11 +30,11 @@ fi
 namespace=$1
 
 tag="latest"
-helm_name="jupyterhub-helm"
+helm_name="jupyterhub-helm-$namespace"
```
[the remark this answers is not in the document]
Comment (privacywill) [reply deployment/jupyterhub/deploy.sh:33]: Deploying the jupyterhub through helm creates a global kubernetes role. If we use the same `helm_name` for every developers even if they belongs to different namespaces, the global role still has conflicts. So I add namespace to the `helm_name`

File: README.md [hunk 2]
```diff
@@ -121,7 +119,7 @@ popd
 
 
 ```shell 
-bazel query 'kind("oci_push", "//...")' | xargs -n1 bazel run
+./push_image.sh --namespace=dcr-namespace
```
[the remark this answers is not in the document]
Comment (privacywill) [reply README.md:122]: done

File: resources/deployment/db_account.tf [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  project  = var.project_id
+  instance = "dcr-${var.env}-db-instance"
+}
+
+resource "google_sql_user" "dcr_db_user" {
+  name     = var.mysql_username
+  instance = "dcr-${var.env}-db-instance"
+  password = var.mysql_password
+  project  = var.project_id
+}
```
[the remark this answers is not in the document]
Comment (privacywill) [reply resources/deployment/db_account.tf:28]: OK

File: resources/deployment/apply.sh [hunk 4]
```diff
@@ -53,11 +71,23 @@ fi
 VAR_FILE=$(realpath $VAR_FILE)
 source $VAR_FILE
 
+cat << EOF > backend.tf
```
[the remark this answers is not in the document]
Comment (privacywill) [reply resources/deployment/apply.sh]: done

**Comment by dayeol** [comment]
made small changes by myself!

**Review by dayeol — APPROVED** [review]
lgtm!

**Pull request merged**